xcodeproj/internal/xcodeproj_rule.bzl

```diff
@@ -480,30 +481,32 @@ actual targets: {}
                 )
 
     # Filter `target_merge_dests` after processing focused targets
-    for dest, src_ids in target_merge_dests.items():
+    for dest, src_targets in target_merge_dests.items():
         if dest not in focused_targets:
             target_merge_dests.pop(dest)
             continue
-        new_srcs_ids = [
-            id
-            for id in src_ids
-            if id in focused_targets
+        new_src_targets = [
+            src_target
+            for src_target in src_targets
+            if src_target.id in focused_targets
         ]
-        if not new_srcs_ids:
+        if not new_src_targets:
             target_merge_dests.pop(dest)
             continue
-        target_merge_dests[dest] = new_srcs_ids
+        target_merge_dests[dest] = new_src_targets
 
     target_merges = {}
     target_merge_srcs_by_label = {}
-    for dest, src_ids in target_merge_dests.items():
-        if len(src_ids) > 1:
-            # We can only merge targets with a single library dependency
+    for dest, src_targets in target_merge_dests.items():
+        focused_src_targets = [src_target for src_target in src_targets if focused_targets.get(src_target.id)]
```

Comment:
Is this check needed? Shouldn't the above `for` loop already filter out ones that aren't focused?

Reply:
Totally right.